javascript/app.js

```diff
@@ -99,11 +99,46 @@ FinancialFreedom.controller('ExpensesInputController', ['$scope', '$location', '
     };
 }]);
 
+FinancialFreedom.directive("btnOutputControl", function($filter){
+    return {
+        scope: {},
+        restrict: 'E',
+        templateUrl: 'partials/output_control.html'
+    }
+});
+
 FinancialFreedom.controller('TimeToRetirementController', ['$scope', 'RetirementCalculatorService', 'CreateRetirementGraphService', function($scope, RetirementCalculatorService, CreateRetirementGraphService) {
     
     var retirement_data = RetirementCalculatorService.calculateRetirementInfo();
+
+    $(function () {
+        $('[data-toggle="popover"]').popover();
+    });
+
     $scope.showSteps = false;
     $scope.penClicked = false;
+
+    $scope.incrementOutputValue = function(output_value, increment) {
+        if (output_value == 'expenses') {
+            $scope.expenses = $scope.expenses + increment;
+        }
+        if (output_value == 'income') {
+            $scope.income = $scope.income + increment;
+        }
+        if (output_value == 'assets') {
+            $scope.assets = $scope.assets + increment;
+        }
+        if (output_value == 'incomeincrease') {
+            $scope.incomeincrease = $scope.incomeincrease + increment;
+        }
+        if (output_value == 'expensesincrease') {
+            $scope.expensesincrease = $scope.expensesincrease + increment;
+        }
+        if (output_value == 'growth') {
+            $scope.growth = $scope.growth + increment;
+        }

```

Comment:
these should probably all be `else if` instead of separate if statements. Also, there might be a better way where instead of passing in `output_value` as a string, pass in the actual model that's supposed to be incremented. I can't remember if angular can do that nicely, though.